### PR TITLE
Remove stable cargo feature `default-run`

### PR DESCRIPTION
**Pull request by lzutao**
At least on beta.

### CODE REVIEW

**Comment by RalfJung** [comment]
We can only do this once the bootstrap-cargo used by rustc also has this as a stable feature. Is that the case?

**Comment by RalfJung** [comment]
Looks like that is
```
$ build/x86_64-unknown-linux-gnu/stage0/bin/cargo --version
cargo 1.37.0-beta (4c1fa54d1 2019-06-24)
```
When did `default-run` get stabilized?

**Comment by lzutao** [comment]
I think it is https://static.rust-lang.org/dist/2019-07-04/rust-std-beta-x86_64-unknown-linux-gnu.tar.gz .
I got warning when building rustc.

**Comment by RalfJung** [comment]
Yeah looks like it is new enough then.  Thanks!

@bors r+

**Comment by RalfJung** [comment]
Actually, this seems to fail on CI. I think CI uses stable cargo currently.

@bors r-

**Comment by lzutao** [comment]
Hm, I should update the rust-toolchain file then.

**Comment by RalfJung** [comment]
You could try changing that to `beta`. The relevant lines are 

https://github.com/rust-lang/miri/blob/3525943794d2fee5b8a55c1a7bd27afceea317e6/.travis.yml#L34

https://github.com/rust-lang/miri/blob/3525943794d2fee5b8a55c1a7bd27afceea317e6/.appveyor.yml#L22

In that case please also open an issue so that we remember to change it back to `stable` when that becomes possible.

**Comment by RalfJung** [comment]
> Hm, I should update the rust-toolchain file then.

No, that has no effect on the cargo we use. It just determines the rustc.

**Comment by lzutao** [comment]
I think it should work now.

**Comment by RalfJung** [comment]
No that will definitely not work. Miri needs lots of updating to work with latest rustc. And even then it will still use stable `cargo`. Please seem my messages above. :)

**Comment by lzutao** [comment]
I would prefer to defer this PR until we sort out issue with latest rustc in
other PRs.

**Comment by RalfJung** [comment]
But why? This is about updating cargo. The other PRs are about updating rustc. Those two are updated entirely independently for our CI.

Fixing this PR is trivial, and I showed you the two lines you will have to change to make that work. :)

**Comment by lzutao** [comment]
Thanks. My assumption is wrong. I opened a tracking issue in #828.

**Comment by lzutao** [comment]
Hm, CI still uses cargo from master to build.

**Comment by RalfJung** [comment]
Hm, either that or it didn't actually install `beta`. The `curl https://build.travis-ci.org/files/rustup-init.sh -sSf | sh -s -- -y --default-toolchain beta` step was suspiciously fast.

**Comment by RalfJung** [comment]
Maybe try removing the `-c cargo` from the `rustup-toolchain-install-master` line?

**Comment by RalfJung** [comment]
Looks good in PR CI. Can you squash, then I will r+?

**Comment by lzutao** [comment]
Done.

File: .travis.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ script:
 
 before_cache:
 # Don't cache "master" toolchain, it's a waste
-- rustup default stable
+- rustup default beta
```
Comment (RalfJung) [inline .travis.yml]: Good catch!

**Comment by RalfJung** [comment]
Great!

@bors r+

**Comment by lzutao** [comment]
Need another bors retry to get it work. Previous fail build only cache beta toolchain.

**Comment by RalfJung** [comment]
I doubt this will work TBH, does the cache get updated when the build fails?

@bors retry

File: .appveyor.yml [hunk 2]
```diff
@@ -19,12 +19,12 @@ cache:
 install:
     # Install Rust
     - curl -sSf -o rustup-init.exe https://win.rustup.rs/
-    - rustup-init.exe -y --default-host %TARGET% --default-toolchain stable
+    - rustup-init.exe -y --default-host %TARGET% --default-toolchain beta
     - set PATH=%USERPROFILE%\.cargo\bin;%PATH%
```
Comment (RalfJung) [inline .appveyor.yml]: If this still fails, you could add `rustup default beta` here. And then once this landed, send a follow-up PR that removes it again.

Comment (lzutao) [reply .appveyor.yml]: Done.

**Comment by RalfJung** [comment]
Turns out I was wrong, switching to beta cargo does not work. It leads to strange failures on Windows: https://github.com/rust-lang/miri/pull/829.

**Comment by RalfJung** [comment]
So given that `rust-version` has been bumped by other PRs, maybe we try your original strategy? I forgot that we did `-c cargo` to actually install the nightly cargo.

**Comment by lzutao** [comment]
Rebased on master.

**Comment by RalfJung** [comment]
Travis looks good!

@bors r+

**Comment by RalfJung** [comment]
Yay, that worked. :D  Thanks!

**Pull request merged**